packages/flutter_tools/test/integration.shard/hot_reload_test.dart

```diff
@@ -35,6 +35,30 @@ void main() {
     await _flutter.hotReload();
   });
 
+  test('multiple overlapping hot reload are debounced and queued', () async {
+    await _flutter.run();
+    // Capture how many *real* hot reloads occur.
+    int numReloads = 0;
+    final StreamSubscription<void> subscription = _flutter.stdout
+        .map(parseFlutterResponse)
+        .where(_isHotReloadCompletionEvent)
+        .listen((_) => numReloads++);
+
+    try {
+      await Future.wait<void>(<Future<void>>[
+        _flutter.hotReload(debounce: true),
```

Comment:
there might be enough latency on CI that we don't hit the 50ms debounce. You could make it configurable (only for the purpose of testing)

Reply:
This code is interacting with a `flutter run` instance, so I don't think that would be easy.

Is your concern that we might see > 50ms between the two "simultaneous" reloads, or that we might see < 50ms between the first set and the second set? (I think we could address the second issue by chaining the 60ms delay onto the end of the initial reloads.. not sure about the first though).

Reply:
The former. It could be configurable via some parameter passed to flutter daemon that is hidden?

Reply:
I've pushed changes that add an argument to override the debounce duration (it's not so much hidden as undocumented). I set the value to 250ms for the test, but let me know if you think it needs to be higher (I'm not sure what a good balance is between avoiding flakes and not making too many slow tests!).

Reply:
250 ms should be good enough for now, but if there are problems with flakiness we can always crank it higher